test-utils/src/main/java/org/hibernate/validator/testutil/ConstraintViolationAssert.java

```diff
@@ -371,6 +401,254 @@ public void containsPaths(PathExpectation... expectedPaths) {
 		}
 	}
 
+	public static class ViolationExpectation {
+
+		private final ViolationExpectationPropertiesToTest propertiesToTest = new ViolationExpectationPropertiesToTest();
+
+		private final Class<? extends Annotation> constraintType;
+
+		private Class<?> rootBeanClass;
+
+		private String message;
+
+		private Object invalidValue;
+
+		private PathExpectation propertyPath;
+
+		public ViolationExpectation(Class<? extends Annotation> constraintType) {
```

Comment:
@gsmet Can we make the constructor private? Only `violationOf()` should be used in tests.